AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/hierarchical-state-machine-swiftTests/StateMachineListenerTests.swift

```diff
@@ -87,27 +87,27 @@ class StateMachineListenerTests: XCTestCase {
 
         wait(for: [subscribed], timeout: 0.1)
 
-        stateMachine.cancel(listenerToken: token)
+        await stateMachine.cancel(listenerToken: token)
         let event = Counter.Event(id: "test", eventType: .increment)
-        stateMachine.send(event)
-        waitForExpectations(timeout: 0.1)
+        await stateMachine.send(event)
+        await waitForExpectations(timeout: 0.1)
     }
 
-    func testDoesNotNotifyIfCancelledImmediately() {
-        stateMachine.send(Counter.Event(id: "test", eventType: .increment))
+    func testDoesNotNotifyIfCancelledImmediately() async {
+        await stateMachine.send(Counter.Event(id: "test", eventType: .increment))
 
         let notified = expectation(description: "notified")
-        notified.isInverted = true
-        let token = stateMachine.listen({ _ in
+        notified.expectedFulfillmentCount = 1
```

Comment:
Note: I think this logic is the right one because, immediately after listening you should get back the current state. Only then cancel should take effect.

Reply:
I agree. Let's get this documented and agreed to across all platforms